### PR TITLE
#909: Copy to clipboard the full path or short name of a current file on small keyboards.

### PR DESCRIPTION
**Pull request by singalen**
Added Cmd/Ctrl-C shortcut to copy current command or file(s), and Alt-Shift-C to copy full path(s).
Not sure this is all that needs to be done, please consider this WIP.

* Is there a plan to resurrect copying files for Explorer/Finder?
* This probably needs to be documented?
* Do we care for UNC paths?

### CODE REVIEW

File: far2l/edit.cpp [hunk 1]
```diff
@@ -1351,7 +1351,9 @@ int Edit::ProcessKey(int Key)
 			Show();
 			return TRUE;
 		}
-		case KEY_CTRLINS:     case KEY_CTRLNUMPAD0:
+		case KEY_CTRLINS:
```
Comment (elfmz) [inline far2l/edit.cpp:1354]: are you sure its needed? There is Key reassignment for KEY_CTRLC: https://github.com/elfmz/far2l/blob/master/far2l/edit.cpp#L695

File: far2l/filelist.cpp [hunk 2]
```diff
@@ -1086,22 +1086,26 @@ int FileList::ProcessKey(int Key)
 			Redraw();
 			return TRUE;
 		case KEY_CTRLINS:      case KEY_CTRLNUMPAD0:
-
-			if (CmdLength>0)
+		case KEY_CTRLSHIFTINS: case KEY_CTRLSHIFTNUMPAD0: case KEY_CTRLC:     // копировать имена
+		case KEY_CTRLALTINS:   case KEY_CTRLALTNUMPAD0:                       // копировать UNC-имена
+		case KEY_ALTSHIFTINS:  case KEY_ALTSHIFTNUMPAD0:  case KEY_ALTSHIFTC: // копировать полные имена
+		    if (CmdLength > 0)
```
Comment (elfmz) [inline far2l/filelist.cpp:1092]: why if (CmdLength > 0) moved?

File: far2l/edit.cpp (hunk 1 above)
Comment (singalen) [reply far2l/edit.cpp:1354]: I am not, will check.

File: far2l/filelist.cpp (hunk 2 above)
Comment (singalen) [reply far2l/filelist.cpp:1092]: To unify the logic: if there is anything in a command line, copy it, and not the selected files. 
OTOH, not all of the shortcuts copy the command line... looks like we need more conditions here.

**Comment by singalen** [comment]
Updated. Not the cleanest patch, but I guess it didn't make it worse.

**Pull request merged**